### PR TITLE
[Settings] Ensure Modern pages update on url changes

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

When testing out some Modern Pages, the modern page components were not updating as a result of url changes. This PR fixes this plus other cleanups.

* Ensure a re-render of Modern pages when the url changes
* Make sure added sections with no configuration are still represented in the Settings data
* Update the `default` query parameter logic so it only applies to legacy pages. We don't want to force the default parameter for modern pages' first section key.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Activate/Enable Gutenberg and turn on the `settings` feature flag
2. WooCommerce > Settings > Shipping. There should now be four sections (there were previously two)
image

3. Add [woocommerce-settings-migration-tester](https://github.com/woocommerce/woocommerce-settings-migration-tester) by running `npm install && npm run plugin-zip` locally and upload. Note that `trunk` has new commits if you've already cloned the repo locally.
4. WooCommerce > Settings > Modern Settings
5. See the tab content should react as you would expect

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Ensure Modern settings pages re-render on updated url

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php [hunk 1]
```diff
@@ -131,12 +131,12 @@ public function add_settings_page_data( $pages ) {
 
 					// Replace empty string section ids with 'default'.
 					$section_id = '' === $section_id ? 'default' : $section_id;
```
Comment (chihsuan) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php]: Looks like this only needed to be run once for each section. Should we also move this outside of loop?

**Review by chihsuan — APPROVED** [review]
LGTM and tested well! 👍  Just left a minor comment.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php]: Good call, did that in https://github.com/woocommerce/woocommerce/pull/53016/commits/131b18daef4721ea1b014ad4c18e47a90a3cf499

**Comment by psealock** [comment]
This is a small change, so I will merge but happy to address any comments @adrianduffell may have in a follow up.

**Comment by adrianduffell** [comment]
> This is a small change, so I will merge but happy to address any comments @adrianduffell may have in a follow up.

All looking good to me 👍

**Pull request merged**